### DISCUSSION TITLE
SchemaGenerator doesn't work with mounted apps that don't implement routes

### DISCUSSION THREAD

**Discussion by and3rson** [encode/starlette#1958] [Potential Issue]
Related issue: https://github.com/mirumee/ariadne/issues/975

Since ariadne doesn't expose any routes but a single `__call__` method, starlette's `SchemaGenerator.get_endpoints` doesn't return any routes for it.

Is it possible to make `SchemaGenerator` call some method (if it exists) on a mounted app (say, `get_openapi_schema() -> str` ~or `get_openapi_endpoints() -> Iterable[EndpointInfo]`~) in order to allow the latter to provide openapi spec for starlette? Alternative option could be to take `__doc__` directly from the mounted app in addition to taking its `.routes`. Not sure if either of these approaches will allow the mounted app to provide spec for more than one endpoint though...

Any suggestions are welcome!

**Comment by rafalp** [comment]
Support for extra argument on `Mount()` that would be a list of dicts with schema contents would be more explicit and leave it to ASGI apps to provide a way to retrieve schema's data by developer integrating those into their project, eg:

```python
graphql = GraphQL(...)

routes = [
  Mount("/graphql", app=graphql, name="graphql", schema=graphql.get_openapi_schema())
]
```

For schema we could do:

```python
class RouteResponseSchema(TypedDict):
    description: str | null
    examples: str | null
    

class RouteSchema(TypedDict):
    path: str
    method: str
    description: str | null
    responses: Dict[int, RouteResponseSchema] | null


Schema = Sequence[RouteSchema]
```
